cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -636,6 +987,58 @@ IndexMap::create_submap(std::span<const std::int32_t> indices) const
   }
 }
 //-----------------------------------------------------------------------------
+std::pair<IndexMap, std::vector<std::int32_t>>
+IndexMap::create_submap_conn(std::span<const std::int32_t> indices) const
+{
+  // Compute the owned, ghost, and ghost owners of submap indices.
+  // NOTE: All indices are local and numbered w.r.t. the original (this) index
+  // map
+  auto [submap_owned, submap_ghost, submap_ghost_owners]
+      = compute_submap_indices(*this, indices);
+
+  // Compute submap offset for this rank
+  std::int64_t submap_local_size = submap_owned.size();
+  std::int64_t submap_offset = 0;
+  int ierr = MPI_Exscan(&submap_local_size, &submap_offset, 1, MPI_INT64_T,
+                        MPI_SUM, _comm.comm());
+  dolfinx::MPI::check_error(_comm.comm(), ierr);
+
+  // Get submap source ranks
+  std::vector<int> submap_src(submap_ghost_owners.begin(),
+                              submap_ghost_owners.end());
+  std::sort(submap_src.begin(), submap_src.end());
+  submap_src.erase(std::unique(submap_src.begin(), submap_src.end()),
+                   submap_src.end());
+  submap_src.shrink_to_fit();
+
+  // Compute submap destination ranks
+  // NOTE: The call to NBX can be avoided by a two-step communication process
```

Comment:
Will you implement this? @IgorBaratta has done some work to remove unnecessary NBX calls, so we don't want to add new unnecessary calls.